internal/metastore/kv/rootcoord/kv_catalog.go

```diff
@@ -417,15 +417,8 @@ func (kc *Catalog) DropCollection(ctx context.Context, collectionInfo *model.Col
 	}
 	// Snapshot will list all (k, v) pairs and then use Txn.MultiSave to save tombstone for these keys when it prepares
 	// to remove a prefix, so though we have very few prefixes, the final operations may exceed the max txn number.
-	// TODO(longjiquan): should we list all partitions & fields in KV anyway?
-	for _, partition := range collectionInfo.Partitions {
-		delMetakeysSnap = append(delMetakeysSnap, BuildPartitionKey(collectionInfo.CollectionID, partition.PartitionID))
-	}
-	for _, field := range collectionInfo.Fields {
-		delMetakeysSnap = append(delMetakeysSnap, BuildFieldKey(collectionInfo.CollectionID, field.FieldID))
-	}
-	// delMetakeysSnap = append(delMetakeysSnap, buildPartitionPrefix(collectionInfo.CollectionID))
-	// delMetakeysSnap = append(delMetakeysSnap, buildFieldPrefix(collectionInfo.CollectionID))
+	delMetakeysSnap = append(delMetakeysSnap, BuildPartitionPrefix(collectionInfo.CollectionID))
```

Comment:
It might easily exceed 128 tnx limitation in etcd if the users enable the partition key function？

Reply:
may exceed